modules/nifti/src/niftireader.cpp

```diff
@@ -112,8 +156,17 @@ const DataFormatBase* NiftiReader::niftiDataTypeToInviwoDataFormat(int niftiData
             return nullptr;
     }
     // clang-format on
-}
+    // In NIFTI-1 files, dimensions 1,2,3 are for space, dimension 4 is for time,
+    // and dimension 5 is for storing multiple values at each spatiotemporal voxel
+    if (niftiImage->dim[5] > 0) {
+        components = niftiImage->dim[5];
+    }
+    if (components > 4) {
+        return nullptr;
```

Comment:
would it make sense to only consider the first 4 components?

Reply:
I would rather give an error here at this point. More than 4 can mean matrices and then it will look like it works even though it is bogus data.